elide-integration-tests/src/test/java/com/yahoo/elide/async/integration/tests/AsyncIT.java

```diff
@@ -111,7 +120,149 @@ public void init() {
     }
 
     /**
-     * Various tests for a JSONAPI query as a Async Request.
+     * Various tests for a JSONAPI query as a Async Request with asyncAfterSeconds value set to 3.
+     * Happy Path Test Scenario 1
+     * @throws InterruptedException
+     */
+    @Test
+    public void jsonApiHappyPath1() throws InterruptedException {
+
+        //Create Async Request
+        given()
+                .contentType(JSONAPI_CONTENT_TYPE)
+                .body(
+                        data(
+                                resource(
+                                        type("asyncQuery"),
+                                        id("edc4a871-dff2-4054-804e-d80075cf830e"),
+                                        attributes(
+                                                attr("query", "/book?sort=genre&fields%5Bbook%5D=title"),
+                                                attr("queryType", "JSONAPI_V1_0"),
+                                                attr("status", "QUEUED"),
+                                                attr("requestId", "1001"),
+                                                attr("asyncAfterSeconds", "3")
+                                        )
+                                )
+                        ).toJSON())
+                .when()
+                .post("/asyncQuery")
+                .then()
+                .statusCode(org.apache.http.HttpStatus.SC_CREATED)
+                .body("data.id", equalTo("edc4a871-dff2-4054-804e-d80075cf830e"))
+                .body("data.type", equalTo("asyncQuery"))
+                .body("data.attributes.status", equalTo("PROCESSING"))
+                .body("data.attributes.result.contentLength", nullValue())
+                .body("data.attributes.result.responseBody", nullValue())
+                .body("data.attributes.result.httpStatus", nullValue())
+                .body("data.attributes.result.resultType", nullValue());
+
+        int i = 0;
+        while (i < 1000) {
+            Thread.sleep(10);
+            Response response = given()
+                    .accept("application/vnd.api+json")
+                    .get("/asyncQuery/edc4a871-dff2-4054-804e-d80075cf830e");
+
+            // If Async Query is created and completed
+            if (response.jsonPath().getString("data.attributes.status").equals("COMPLETE")) {
+
+                // Validate AsyncQuery Response
+                response
+                        .then()
+                        .statusCode(HttpStatus.SC_OK)
+                        .body("data.id", equalTo("edc4a871-dff2-4054-804e-d80075cf830e"))
+                        .body("data.type", equalTo("asyncQuery"))
+                        .body("data.attributes.queryType", equalTo("JSONAPI_V1_0"))
+                        .body("data.attributes.status", equalTo("COMPLETE"));
+
+                // Validate AsyncQueryResult Response
+                given()
```

Comment:
No need to issue another query here - combine the response body checking with body assertions above.